src/main/java/ch/naviqore/service/Leg.java

```diff
@@ -0,0 +1,22 @@
+package ch.naviqore.service;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public interface Leg {
+    @NotNull Location getDepartureLocation();
+
+    @NotNull Location getArrivalLocation();
```

Comment:
I would not return the location separately... the location can be retrieved via the stop.getLocation()

Reply:
A leg can be from a random coordinate to the nearest stop. So there are cases where stop will be null